### PR TITLE
[94x] Run2_HI modifier chain use stage2L1Trigger

### PR DESCRIPTION
**Pull request by rekovic**
For Run2_HI modifier chain, use stage2L1Trigger_2017 instead of stage1L1Trigger.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

File: Configuration/Eras/python/Era_Run2_HI_cff.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from Configuration.Eras.Modifier_run2_common_cff import run2_common
 from Configuration.Eras.Modifier_run2_HI_specific_cff import run2_HI_specific
-from Configuration.Eras.Modifier_stage1L1Trigger_cff import stage1L1Trigger
+from Configuration.Eras.Modifier_stage2L1Trigger_2017_cff import stage2L1Trigger_2017
 
-Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage1L1Trigger)
+Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage2L1Trigger_2017)
```
Comment (slava77) [inline Configuration/Eras/python/Era_Run2_HI_cff.py]: this era is still supposed to work with 2015 HI data.
Is the "2017" here simply nominal and will still work on that data?

Comment (rekovic) [reply Configuration/Eras/python/Era_Run2_HI_cff.py]: @slava77 
In that case we better use stage2L1Trigger modifier.

Comment (rekovic) [reply Configuration/Eras/python/Era_Run2_HI_cff.py]: Will provide the change and retest.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
Also checked that wf 145.0 is passing.

With this PR merged, we can again use L1 NoHackConditions PR, i.e. revert the revert PR #20921,  which was done due to problematic wf 145.0.

**Comment by arunhep** [comment]
@rekovic @davidlange6 
will merging this PR solve the failure of wf145.0? If yes, then pls merge it.

File: Configuration/Eras/python/Era_Run2_HI_cff.py (hunk 1 above)
Comment (davidlange6) [reply Configuration/Eras/python/Era_Run2_HI_cff.py]: but did we have the stage 2 trigger in 2015? @rekovic

**Comment by rekovic** [comment]
@arunhep 
> will merging this PR solve the failure of wf145.0?

Yes. This PR uses L1T Stage2 instead of Stage1 in case of Run2_HI modifier. 
Apart from standard jenkins test, additional tests successfully performed:
- wf 145.0 passes with this PR
- wf 145.0 passes with this PR plus NoHackConditions (i.e. Revert "Revert "[94X] L1 no hack conditions - cleaned"" #20957)

@davidlange6 Please merge this PR at your earliest convenience.  And then can merge #20957 immediately after.

**Comment by davidlange6** [comment]
as already asked

>  
-Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage1L1Trigger)
+Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage2L1Trigger_2017)

but did we have the stage 2 trigger in 2015? @rekovic

—

**Comment by rekovic** [comment]
@davidlange6 
In fact, the change is:
-Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage1L1Trigger)		 
+Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage2L1Trigger)

So the switch is to stage2L1Trigger.  
Answer to your question is, no we did not run stage2 trigger in 2015, but emulation of stage2 trigger on 2015 data works as well.

**Comment by davidlange6** [comment]
so this this change is not correct.. we want a trigger that corresponds to what was run in 2015 PbPb run when doing 2015PbPb processing. 

But lets jump back - the workflow in question is a 2015 HI workflow. So, the ERA is correct. I take it that the conditions are wrong?

**Comment by rekovic** [comment]
@davidlange6 OK.  In that case the conditions need to be changed. 

@kkotov @arunhep Please revert queue 94X_mcRun2_HeavyIon to previous tag.

**Comment by kkotov** [comment]
@rekovic , @arunhep 
I don't know if the 94X_mcRun2_HeavyIon global tag is also foreseen for some heavy ion runs of next year. If so, rolling conditions back may just postpone this problem till we update the CaloL2 conditions and have Stage1 crashing again. There are several other options available assuming that Stage1 conditions is a history and cannot change:
1) resort to using static pythons for Stage1 as we had for Stage2 before (an ugly alternative)
2) plug into Stage1 pythons an "es_prefered" PoolESSource that explicitly queries Stage1 conditions
3) resurrect Mike's design and have Stage1 and Stage2 conditions in the same GTs linked to different records (that way we bump into crashing HLT tests again)
4) follow Arun's proposal and cook up a frozen Stage1 GT that makes happy all of the legacy workflows

I personally like the last option (4). If everyone on board, let's leave the current HI GT as is and push for a new Stage1 GTs.

**Comment by davidlange6** [comment]
yes, in 2018 there will be a pbpb run - so that will have the stage2 trigger and a different alca auto GT that will include the stage 2 payloads. Normal procedure.

**Comment by rekovic** [comment]
closing this PR, and making the change in GTs to make the wf145.0 all in the PR of revert-revert-NoHackConditions #20957.

**Pull request closed without merging**